modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientHandlerModule.java

```diff
@@ -276,7 +275,7 @@ public void enable() {
      */
     private CompletableFuture<Channel> startEndpoint(ClientConnectorView configuration) {
         ServerBootstrap bootstrap = bootstrapFactory.createServerBootstrap();
-        CompletableFuture<ClusterTag> clusterTag = clusterTagSupplier.get();
+        CompletableFuture<ClusterInfo> clusterInfo = clusterInfoSupplier.get();
```

Comment:
`startEndpoint` is called once during node startup. Looks like we won't receive any cluster id history updates that may occur during the node life time.

Reply:
Currently, this doesn't seem to be a problem as a cluster ID is always changed via a restart. But we now have a storage from which we can read ClusterState, allowing us to not use the future (and `join()`) anymore, so I switched to this approach.